File: android/app/src/main/java/com/anonymous/firstApp/MyNativeModule.kt
package com.anonymous.firstApp

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class MyNativeModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    private var sharedVariable: String? = null

    override fun getName(): String {
        return "MyNativeModule"
    }

    @ReactMethod
    fun receiveVariable(variable: String) {
        sharedVariable = variable
    }

    @ReactMethod
    fun getSharedVariable(promise: Promise) {
        promise.resolve(sharedVariable)
    }
}

File: ios/MyNativeModule.swift
import Foundation

@objc(MyNativeModule)
class MyNativeModule: NSObject {

  private var sharedVariable: String?

  @objc
  func receiveVariable(_ variable: String) {
    sharedVariable = variable
  }

  @objc
  func getSharedVariable(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    if let variable = sharedVariable {
      resolve(variable)
    } else {
      reject("no_variable", "No variable received", nil)
    }
  }
}
